avrohugger-core/src/test/scala/scavro/ScavroStringToStringsSpec.scala

```diff
@@ -150,6 +145,16 @@ class ScavroStringToStringsSpec extends mutable.Specification {
       source === expected
     }
 
+    "17. correctly generate with messy schema" in {
+      val inputString = util.Util.readFile("avrohugger-core/src/test/avro/clash.avsc")
+      val gen = new Generator(Scavro)
+      val List(source0, source1, source2) = gen.stringToStrings(inputString)
+
+      source0 === util.Util.readFile("avrohugger-core/src/test/expected/scavro/example/model/ClashInner.scala")
+      source1 === util.Util.readFile("avrohugger-core/src/test/expected/scavro/example/model/ClashOuter.scala")
+      source2 === util.Util.readFile("avrohugger-core/src/test/expected/scavro/example/model/ClashRecord.scala")
```

Comment:
This is the one that will give invalid scala, calling `new JClashInner()` or `ClashInner()` without the argument list.